boost-maven-plugin/src/main/java/boost/project/LibertyBoostMojo.java

```diff
@@ -66,22 +72,34 @@
     private ArtifactItem runtimeArtifact;
 
     public void execute() throws MojoExecutionException {
+
+    	List<String> deps = null;
+    	boosterParent = new BoosterPacksParent();
+    
         createDefaultRuntimeArtifactIfNeeded();
 
         createServer();
-
-        try {
-            generateServerXML();
-        } catch (TransformerException | ParserConfigurationException e) {
+    	
+        String springBootVersion = SpringBootProjectUtils.findSpringBootVersion(mavenProject);
+        
+		try {
+			if (springBootVersion != null) {
+				// dealing with a spring boot app
+				installApp("spring-boot-project");
+				generateServerXML();
+				generateBootstrapProps();
+			} else {
+				// dealing with an EE based app
+				installApp("project");
+				boosterFeatures = getBoosterConfigsFromDependencies(mavenProject);
+				generateServerXMLJ2EE(boosterFeatures);
```

Comment:
We can probably do this in another item, but might be good to eventually have a generic config generator that can work for either the spring or the j2ee case.

Reply:
yes, definitely agree on that - Scott and I started discussing that yesterday, but felt i should get this stake in the ground first and then work on that